src/components/modal/modals/index.ts

```diff
@@ -0,0 +1,18 @@
+import { DeleteItemModal, DeleteItemModalProps } from './DeleteItemModal';
```

Comment:
This can just be:
```
import { DeleteItemModal } from './DeleteItemModal';
import { UpgradeDriveModal } from './UpgradeDriveModal';

export const modals = {
    deleteItem: DeleteItemModal,
    upgradeDrive: UpgradeDriveModal,
} as const;

export type Modals = typeof modals;

export type ModalType = keyof Modals;

export type ModalPropsMapping = {
    [K in ModalType]: React.ComponentProps<Modals[K]>;
};

```

This way we don't need to define `ModalPropsMapping` manually.

Reply:
nice catch!
included here: [b63af9b](https://github.com/powerhouse-inc/document-model-electron/pull/31/commits/b63af9b69bf4e4f535ba2532ad5f86ffcc727169)